scripts/linters/codeowner_linter_test.py

```diff
@@ -0,0 +1,223 @@
+
+        codeowner_important_paths_swap = self.swap(
+            codeowner_linter, 'CODEOWNER_IMPORTANT_PATHS',
+            mock_codeowner_important_paths)
+
+        with self.print_swap, codeowner_important_paths_swap:
+            codeowner_linter.check_codeowner_file(FILE_CACHE, True)
+        self.assertTrue(
+            appears_in_linter_stdout(
+                ['.github/CODEOWNERS --> Rule /.github/stale.yml is not present'
+                 ' in the CODEOWNER_IMPORTANT_PATHS list in scripts/linters/'
+                 'pre_commit_linter.py. Please add this rule in the '
+                 'mentioned list or remove this rule from the \'Critical files'
+                 '\' section.'], self.linter_stdout))
+
+    def test_duplicate_important_patterns_at_the_bottom_of_codeowners(self):
+        codeowner_path_swap = self.swap(
+            codeowner_linter, 'CODEOWNER_FILEPATH', MOCK_CODEOWNER_FILEPATH)
+
+        with self.print_swap, codeowner_path_swap:
+            codeowner_linter.check_codeowner_file(FILE_CACHE, True)
+        self.assertTrue(
+            appears_in_linter_stdout(
+                ['Duplicate pattern(s) found in critical '
+                 'rules section.'], self.linter_stdout))
+
+    def test_duplicate_important_patterns_in_list(self):
+        mock_codeowner_important_paths = (
+            CODEOWNER_IMPORTANT_PATHS + [CODEOWNER_IMPORTANT_PATHS[-1]])
+
+        codeowner_important_paths_swap = self.swap(
+            codeowner_linter, 'CODEOWNER_IMPORTANT_PATHS',
+            mock_codeowner_important_paths)
+
+        with self.print_swap, codeowner_important_paths_swap:
+            codeowner_linter.check_codeowner_file(FILE_CACHE, True)
+        self.assertTrue(
+            appears_in_linter_stdout(
+                ['Duplicate pattern(s) '
+                 'found in CODEOWNER_IMPORTANT_PATHS list.'],
+                self.linter_stdout))
+
+    def test_extra_important_patterns_in_important_pattern_list(self):
+        codeowner_path_swap = self.swap(
+            codeowner_linter, 'CODEOWNER_FILEPATH', MOCK_CODEOWNER_FILEPATH)
+
+        with self.print_swap, codeowner_path_swap:
+            codeowner_linter.check_codeowner_file(FILE_CACHE, True)
+        self.assertTrue(
+            appears_in_linter_stdout(
+                ['Rule \'/.github/stale.yml\' is not present in '
+                 'the \'Critical files\' '
+                 'section. Please place it under the \'Critical files\' '
+                 'section since it is an important rule. Alternatively please '
+                 'remove it from the \'CODEOWNER_IMPORTANT_PATHS\' list in '
+                 'scripts/linters/pre_commit_linter.py if it is no longer an '
+                 'important rule.'], self.linter_stdout))
+
+    def test_check_codeowner_file_with_success_message(self):
+        with self.print_swap:
+            codeowner_linter.check_codeowner_file(FILE_CACHE, False)
```

Comment:
are we using the original codeowner file here?

Reply:
Yes

Reply:
Should we avoid using the original file for the lint test? Can you explain why can't we use the dummy CODEOWNER file to test the success message? [I mean this would again be like running the CODEOWNER checks using test files isn't it?]

Reply:
Used dummy file.